scipy/signal/windows/_windows.py

```diff
@@ -37,7 +45,7 @@ def _truncate(w, needed):
         return w
 
 
-def general_cosine(M, a, sym=True):
+def general_cosine(M, a, sym=True, *, array_namespace=None):
```

Comment:
WDYT about naming the parameter `xp` instead? It is obviously a little more vague, but I feel that `array_namespace` is a bit too verbose for public API. That would align with what we already have in, e.g., [`fftfreq`](https://docs.scipy.org/doc/scipy/reference/generated/scipy.fft.fftfreq.html#scipy.fft.fftfreq)

Reply:
along the same comparison, perhaps we would want to add a `device` kwarg here too.

Reply:
Funny, I originally named the parameter `xp`, but thought it was a bit too vague for a public API, so I did a search-and-replace 😁

Reply:
Happy to change it back if you prefer that.

Reply:
Done